platform/darwin/docs/guides/Style Layers Using Expressions.md.ejs

```diff
@@ -0,0 +1,168 @@
+
+<%- guideExample(guide, 'Categorical', os) %>
+
+`MGL_MATCH` takes the attribute key, followed by possible attribute and style value pairs. The final argument is a default style value.
+
+![categorical mode](img/data-driven-styling/categorical1.png) ![categorical mode](img/data-driven-styling/categorical2.png)
+
+### Identity
+
+Identity interpolation mode uses the attribute’s value as the style layer property value. For example, you can set the `circleRadius` to the earthquake’s magnitude. Since the attribute value itself will be used as the style value, `sourceStops` should be set to `nil`.
+
+<%- guideExample(guide, 'Identity', os) %>
+
+![identity mode](img/data-driven-styling/identity.png)
+
+Some built-in expressions functions can be applied to attribute values to style layer property values. To set the circle radius to 3 times the earthquake’s magnitude, create a `multiply:by` function that takes the attribute value and the multiplier as arguments.
+
+<%- guideExample(guide, 'Multiply', os) %>
+
+![multiply magnitude](img/data-driven-styling/multiply.png)
+
+Through expressions, you can also cast attribute values in order to use them. One example is to cast an integer as an `NSString` and use it as a text value.
+
+<%- guideExample(guide, 'Cast', os) %>
+
+![cast a value](img/data-driven-styling/cast.png)
+
+<% if (iOS) { -%>
+
+  ##Migrating from Style Functions to Expressions
+
+  For constant values that do not necessarily change based on camera or attribute values, use `[NSExpression expressionForConstantValue:]` (previously `[MGLStyleValue valueWithRawValue:]`). Expressions support format string syntax, as shown above, as well as convenience methods that include those listed below.
+
+
+<table>
+<thead>
+<tr><th>Interpolation mode</th><th>Syntax prior to v4.0.0</th><th>Current syntax</th></tr>
+</thead>
+<tbody>
+<tr>
+  <td>Linear </td>
+  <td><code>MGLStyleValue(interpolationMode: .exponential,
+                                        sourceStops: stops,
+                                        attributeName: "mag",
+                                        options: nil) </code></td>
+  <td><code>NSExpression(forMGLInterpolating: NSExpression(forKeyPath: "mag"),
+                  curveType: MGLExpressionInterpolationMode.linear,
+                  parameters: nil,
+                  stops: NSExpression(forConstantValue: stops)) </code></td>
+</tr>
+<tr>
+  <td>Exponential</td>
+  <td><code>MGLStyleValue(interpolationMode: .exponential,
+                                        sourceStops: stops,
+                                        attributeName: “mag”,
+                                         options:  [.interpolationBase: 1.5])</code></td>
+  <td><code>layer.circleRadius = NSExpression(format: "FUNCTION($zoomLevel, 'mgl_interpolateWithCurveType:parameters:stops:', 'exponential', 1.5, %@)",
+                                          stops)</code></td>
```

Comment:
If the sections above focus on migration and already contain these code blocks, then this summary table can be condensed for readability. For example:

Interpolation mode | Syntax prior to v4.0.0 | Current syntax
----|----|----
Exponential | Pass `MGLInterpolationMode.exponential` into `MGLStyleValue(interpolationMode:sourceStops:attributeName:options:)` | Pass `NSExpression.zoomLevelVariableExpression` into `+[NSExpression(MGLInitializerAdditions) mgl_expressionForInterpolatingExpression:withCurveType:parameters:stops:]`

Another benefit is that we won’t need to test the code snippets within the table.